language-support/ts/codegen/tests/src/DA/Test/Daml2Ts.hs

```diff
@@ -85,22 +85,23 @@ tests rootDir yarn damlc daml2ts davl = testGroup "daml2Ts"
       assertBool "grover-1.0.dar was not created." =<< doesFileExist groverDar
       step "Generating TypeScript of 'grover'..."
       daml2tsProject [groverDar] (groverTs </> "src")
-      assertBool "'Grover.ts' was not created." =<< doesFileExist (groverTs </> "src" </> "grover-1.0" </> "Grover.ts")
-      assertBool "'packageId.ts' was not created." =<< doesFileExist (groverTs </> "src" </> "grover-1.0" </> "packageId.ts")
+      assertBool "'Grover.ts' was not created." =<< doesFileExist (groverTsSrc </> "grover-1.0" </> "Grover.ts")
+      assertBool "'packageId.ts' was not created." =<< doesFileExist (groverTsSrc </> "grover-1.0" </> "packageId.ts")
       step "Compiling 'grover-ts' to JavaScript... "
-      writeTsConfig groverTs
-      writePackageJson groverTs "@breathing-test/grover"
-      writeRootPackageJson rootDir "breathing-test/grover-ts"
-      withCurrentDirectory rootDir $ yarnProject ["workspaces", "run", "build"]
-      assertBool "'Grover.js' was not created." =<< doesFileExist (groverTs </> "lib" </> "grover-1.0" </> "Grover.js")
+      writePackageConfigs groverTs
+      withCurrentDirectory groverTs $ yarnProject ["run", "build"]
+      assertBool "'Grover.js' was not created." =<< doesFileExist (groverTsLib </> "grover-1.0" </> "Grover.js")
+      step "Linting 'grover-ts' ... "
+      withCurrentDirectory groverTs $ yarnProject ["run", "lint"]
```

Comment:
What do you think about putting the calls to `step` and `yarnProject` into a function `yarnLint`. This combination appears quite a few times.

Reply:
I think it's ameliorated by the earlier suggestion of making better use of `withCurrentDirectory`.